dotnet/src/Connectors/Connectors.Memory.SqlServer.Classic/Core/SqlServerClient.cs

```diff
@@ -0,0 +1,473 @@
+using System.Threading.Tasks;
+
+namespace Microsoft.SemanticKernel.Connectors.SqlServer.Classic.Core;
+
+/// <summary>
+/// Represents a client for interacting with a SQL Server database for storing semantic memories and embeddings.
+/// </summary>
+[System.Diagnostics.CodeAnalysis.SuppressMessage("Security", "CA2100:Review SQL queries for security vulnerabilities",
+                                                   Justification = "We need to build the full table name using schema and collection, it does not support parameterized passing.")]
+internal sealed class SqlServerClient
+{
+    private readonly string _connectionString;
+    private readonly SqlServerConfig _configuration;
+    private SqlConnection _connection;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SqlServerClient"/> class with the specified connection string and schema.
+    /// </summary>
+    /// <param name="connectionString">The connection string to use for connecting to the SQL Server database.</param>
+    /// <param name="configuration">The configuration to use for the SQL Server database.</param>
+    public SqlServerClient(string connectionString, SqlServerConfig configuration)
+    {
+        this._connectionString = connectionString;
+        this._configuration = configuration;
+
+        this._connection = new SqlConnection(this._connectionString);
+        this._connection.Open();
+    }
+
+    /// <inheritdoc />
+    public async Task CreateTablesAsync(CancellationToken cancellationToken)
+    {
+        var sql = $@"IF NOT EXISTS (SELECT  *
+                                    FROM    sys.schemas
+                                    WHERE   name = N'{this._configuration.Schema}' )
+                    EXEC('CREATE SCHEMA [{this._configuration.Schema}]');
+                    IF OBJECT_ID(N'{this.GetFullTableName(this._configuration.MemoryCollectionTableName)}', N'U') IS NULL
+                    CREATE TABLE {this.GetFullTableName(this._configuration.MemoryCollectionTableName)}
+                    (   [id] NVARCHAR(256) NOT NULL,
+                        PRIMARY KEY ([id])
+                    );
+
+                    IF OBJECT_ID(N'{this.GetFullTableName(this._configuration.MemoryTableName)}', N'U') IS NULL
+                    CREATE TABLE {this.GetFullTableName(this._configuration.MemoryTableName)}
+                    (   [id] UNIQUEIDENTIFIER NOT NULL,
+                        [key] NVARCHAR(256)  NOT NULL,
+                        [collection] NVARCHAR(256) NOT NULL,
+                        [metadata] TEXT,
+                        [embedding] TEXT,
+                        [timestamp] DATETIMEOFFSET,
+                        PRIMARY KEY ([id]),
+                        FOREIGN KEY ([collection]) REFERENCES {this.GetFullTableName(this._configuration.MemoryCollectionTableName)}([id]) ON DELETE CASCADE,
+                        CONSTRAINT UK_{this._configuration.MemoryTableName} UNIQUE([collection], [key])
```

Comment:
Unless I missed something, there's mostly no escaping for these identifiers (in general, not only this constraint). This should be added.

Reply:
Added escaped characters.